### ISSUE TITLE
setup.py installs systemd drop-ins to /etc, should be /usr/lib

### ISSUE THREAD

**Issue by nmeyerhans** [canonical/cloud-init#5613]
# Bug report

Forwarded from https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1078180

## Steps to reproduce the problem

The setup.py [code](https://github.com/canonical/cloud-init/blob/4c96055e5c714f5cb2bc709f410dcd495bb9f330/setup.py#L311-L319) installs the `systemd/disable-sshd-keygen-if-cloud-init-active.conf` file to `/etc/systemd/system` using the following code:

```python
    data_files.extend(
        [
            (RULES_PATH + "/rules.d", [f for f in glob("udev/*.rules")]),
            (
                INITSYS_ROOTS["systemd"] + "/sshd-keygen@.service.d/",
                ["systemd/disable-sshd-keygen-if-cloud-init-active.conf"],
            ),
        ]
    )
```

As noted in the Debian bug report, the correct location for this file is the distribution directory, `/usr/lib`.  See also the [UNIT FILE LOAD PATH](https://manpages.debian.org/bookworm/systemd/systemd.unit.5.en.html#UNIT_FILE_LOAD_PATH) section of the systemd documentation, which notes that /etc is reserved for "System units created by the administrator" while the lib directory should be used by "System units installed by the distribution package manager".

In other words, cloud-init currently installs 
`/etc/systemd/system/sshd-keygen@.service.d/disable-sshd-keygen-if-cloud-init-active.conf`
which should instead be installed as 
`/usr/lib/systemd/system/sshd-keygen@.service.d/disable-sshd-keygen-if-cloud-init-active.conf`
